### ISSUE TITLE
[BUG] speed not controlled JTX Sprint-5

### ISSUE THREAD

**Issue by Jiffy176** [cagnulein/qdomyos-zwift#2223]
**Describe the bug**

**To Reproduce**
1. Go to '...'app doesn't control treadmill speed only elevation when running

**Expected behavior**
A clear and concise description of what you expected to happen.increase/ decrease speed

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
[debug-Thu_Mar_14_10_20_45_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14601018/debug-Thu_Mar_14_10_20_45_2024.log)
[debug-Thu_Mar_14_10_27_25_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14601019/debug-Thu_Mar_14_10_27_25_2024.log)

**Additional context**

**Comment by cagnulein** [comment]
hi @Jiffy176 yes your treadmill is not capable to change speed via bluetooth, only inclination is available due to safety reason. If you are aware of any app that can control the speed too in case I can mimic it, but as far as i can tell you, your treadmill doesn't have this capability

**Comment by cagnulein** [comment]
@Jiffy176 yes enable the android notification icon in the experimental settings. you can also have the floating window in overlay to peloton on the same device from qz!

**Comment by Jiffy176** [comment]
How do I do the floating window?

**Comment by cagnulein** [comment]
Click on the icon on the top left corner of qz!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 14 mar 2024 alle 17:35 Jiffy176 ***@***.***> ha
scritto:

> How do I do the floating window?
>

**Comment by Jiffy176** [comment]
Ok next question.
When using a galaxy S24. I open the app and wait for it to connect,
Then open peloton which overlays the qd app and as such it disconnects from
the tread. Any ideas please?

**Comment by cagnulein** [comment]
did you enable the android notification icon in the experimental settings as I suggested?

**Comment by Jiffy176** [comment]
Yes that was what surprised me

**Comment by cagnulein** [comment]
did you see the QZ notification?

**Comment by Jiffy176** [comment]
Yes

**Comment by cagnulein** [comment]
check your power save settings for qz, maybe samsung is killing it because of them

**Comment by Jiffy176** [comment]
Will do

**Comment by Jiffy176** [comment]
Hi Roberto
Sorry more issues.
So I got both apparently running on the same phone. You were right battery
saver issues.
However now everyone I increase incline it cuts out and the treadmill
stops.

It doesn't do this on decline only incline when I press the + on the app
tile and or when peloton class tries to increase the incline.
Any ideas on how to sort it.
Just trying to get a debug log but struggling to find it.
Thanks.
S

**Comment by cagnulein** [comment]
It could be an hardware issue if it doesn't on decline. Which is the issue
with the debug log?
You already did it once

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 15 mar 2024 alle 20:35 Jiffy176 ***@***.***> ha
scritto:

> Hi Roberto
> Sorry more issues.
> So I got both apparently running on the same phone. You were right battery
> saver issues.
> However now everyone I increase incline it cuts out and the treadmill
> stops.
>
> It doesn't do this on decline only incline when I press the + on the app
> tile and or when peloton class tries to increase the incline.
> Any ideas on how to sort it.
> Just trying to get a debug log but struggling to find it.
> Thanks.
> S
>

**Comment by Jiffy176** [comment]
Yeah I'm on a different phone android 14 can't find the log using either
apps Samsung s24??

**Comment by cagnulein** [comment]
For android 14 you need the beta so join here
https://play.google.com/apps/testing/org.cagnulen.qdomyoszwift
Then update the version and you will find the logs in  /documents/qz

Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 15 mar 2024 alle 20:44 Jiffy176 ***@***.***> ha
scritto:

> Yeah I'm on a different phone android 14 can't find the log using either
> apps Samsung s24??
>

**Comment by Jiffy176** [comment]
Cool waiting for the update thanks

**Comment by Jiffy176** [comment]
I've sent over the logs.
If the tread won't allow control of incline. This issue is random
Is there a setting to stop the app asking each time I start a peloton run?
Thanks
S

**Comment by cagnulein** [comment]
you have to attach the log from github web interface directly otherwise i can't see them
for peloton, just remove your peloton credentials if you don't want it and press ok

**Comment by Jiffy176** [comment]
[debug-Fri_Mar_15_18_36_14_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620806/debug-Fri_Mar_15_18_36_14_2024.log)
[debug-Fri_Mar_15_18_37_37_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620807/debug-Fri_Mar_15_18_37_37_2024.log)
[debug-Fri_Mar_15_18_38_56_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620808/debug-Fri_Mar_15_18_38_56_2024.log)
[debug-Fri_Mar_15_18_40_07_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620809/debug-Fri_Mar_15_18_40_07_2024.log)
[debug-Fri_Mar_15_19_04_03_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620811/debug-Fri_Mar_15_19_04_03_2024.log)
[debug-Fri_Mar_15_19_06_02_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620814/debug-Fri_Mar_15_19_06_02_2024.log)
[debug-Fri_Mar_15_19_07_11_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620815/debug-Fri_Mar_15_19_07_11_2024.log)
[debug-Fri_Mar_15_19_08_22_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14620816/debug-Fri_Mar_15_19_08_22_2024.log)


Logs attached. Hopefully cheers.

**Comment by cagnulein** [comment]
ok i'm going to bed now, I will check them ASAP

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 15 mar 2024 alle ore 22:15 Jiffy176 ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
@Jiffy176 do you have any other app that works fine with your treadmill?

**Comment by Jiffy176** [comment]
Yes the kinomap app only

**Comment by Jiffy176** [comment]
No problem
 Cheers for such a quick response

**Comment by cagnulein** [comment]
ok so with kinomap can you change speed and incline without any issue? are you able using it to follow this https://github.com/cagnulein/qdomyos-zwift/blob/master/docs/20_supported_devices_and_applications.md#android-device 
 ? so i will see which is the difference

**Comment by Jiffy176** [comment]
Will check tmrw thanks.

**Comment by Jiffy176** [comment]
Kinomap only controls the incline  not speed. Will try what you sent
tomorrow and see what I can get for you. Thanks.

**Comment by cagnulein** [comment]
ok so probably that's why. i guess speed is not controllable for your treadmill.

if you don't change speed on qz everything is fine?

**Comment by Jiffy176** [comment]
No the speed on qz does  nothing as expected  so I haven't touched that.
When I incline on qz that stops the run.
Qz measures speed and incline ok and reacts when I adjust on the machine
just the elevation control that is a problem.
Thanks.

**Comment by cagnulein** [comment]
ok so with the kinomap log i should be able to see if they did something different 

thanks

**Comment by Jiffy176** [comment]
Sorry stupid question.
I have put my phone in developer mode and enabled the Bluetooth hci snoop
and restart.
What app do I use to see if the data is reading the tried Kino or qd or
another?
Thanks.

**Comment by cagnulein** [comment]
yes just kinomap without qz

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle 10:14 Jiffy176 ***@***.***> ha
scritto:

> Sorry stupid question.
> I have put my phone in developer mode and enabled the Bluetooth hci snoop
> and restart.
> What app do I use to see if the data is reading the tried Kino or qd or
> another?
> Thanks.
>

**Comment by Jiffy176** [comment]
[btsnoop_hci.log](https://github.com/cagnulein/qdomyos-zwift/files/14623160/btsnoop_hci.log)
I think this is the BT log you need. 
I was using Kinomap on a mountain run where the incline and decline of the run was controlled by the app and it all worked fine. 
Thanks.

**Comment by Jiffy176** [comment]
Should be with you now

**Comment by cagnulein** [comment]
This is perfect, thank you!
i will check it asap (probably next week)

Il giorno sab 16 mar 2024 alle 11:07 Jiffy176 ***@***.***> ha
scritto:

**Comment by Jiffy176** [comment]
Cool thanks so much

**Comment by cagnulein** [comment]
hi @Jiffy176 i was quickly checking your log.
Try to enable the "DKN Endurn Treadmill" in the toorx settings in qz and restart QZ.
Let me know if it's better!

**Comment by cagnulein** [comment]
same issue here also https://github.com/cagnulein/qdomyos-zwift/issues/1762

**Comment by Jiffy176** [comment]
Ooo that's interesting did it ever get resolved?

**Comment by Jiffy176** [comment]
What setting should I have speed and incline set to I have incline 1.0
Thanks

**Comment by cagnulein** [comment]
check this https://github.com/cagnulein/qdomyos-zwift/issues/2223#issuecomment-2001994910

**Comment by Jiffy176** [comment]
No joy I'm afraid when changing to dkn the app doesn't read any thing of change the incline.

**Comment by cagnulein** [comment]
could please make me a debug log from qz in this scenario just to be sure?
Thanks!

**Comment by Jiffy176** [comment]
Can't upload for some reason

**Comment by cagnulein** [comment]
zip or put in a google driver

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle ore 15:29 Jiffy176 ***@***.***>
ha scritto:

**Comment by Jiffy176** [comment]
[debug-Sat_Mar_16_14_25_29_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14623895/debug-Sat_Mar_16_14_25_29_2024.log)

**Comment by Jiffy176** [comment]
Had to copy to Google drive hope that was ok

**Comment by cagnulein** [comment]
unfortunately it's missing the first and most important part.
are you able to do it again?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle ore 15:33 Jiffy176 ***@***.***>
ha scritto:

**Comment by Jiffy176** [comment]
I can but I'm not sure what it would show as the only thing that happens is
the tread connects to the app. It displays nothing on the tiles.  Does that
help?

**Comment by cagnulein** [comment]
yes it would help anyway because the communication is anyway in progress on the bluetooth side

**Comment by Jiffy176** [comment]
Ok

**Comment by cagnulein** [comment]
in any case I will check the hci snoof in order to match the differences

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle ore 15:51 Jiffy176 ***@***.***>
ha scritto:

> Ok
>

**Comment by Jiffy176** [comment]
[debug-Sat_Mar_16_15_10_50_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14624125/debug-Sat_Mar_16_15_10_50_2024.log)

Of note the treadmill beeps twice when it connects to the phone. With the DKN setting it does not beep despite the app saying connected. 
Thanks.

**Comment by cagnulein** [comment]
ok as i guessed, got this issue.
let me try to fix this now

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle ore 16:15 Jiffy176 ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
done! https://github.com/cagnulein/qdomyos-zwift/actions/runs/8308710354
Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.
Let me know if you have any questions

**Comment by cagnulein** [comment]
also remember to enable back the "DKN Endurn Treadmill" setting.
This is just a try, in case it work i will push it to the play store.

**Comment by Jiffy176** [comment]
Ok just out at the moment.
So I delete current app
Download android link and open that one and then see how it goes once I've
enabled the dkn bit?
Is that correct. Thanks.

**Comment by cagnulein** [comment]
Exactly

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle 17:01 Jiffy176 ***@***.***> ha
scritto:

> Ok just out at the moment.
> So I delete current app
> Download android link and open that one and then see how it goes once I've
> enabled the dkn bit?
> Is that correct. Thanks.
>

**Comment by Jiffy176** [comment]
Hi.
That didn't work.
I get one beep and that's it when the app connects and all the keys on the
tread are disabled.
Will send log

**Comment by cagnulein** [comment]
ok thanks, so maybe there is also something else. a log will highlight the
difference

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle 20:38 Jiffy176 ***@***.***> ha
scritto:

> Hi.
> That didn't work.
> I get one beep and that's it when the app connects and all the keys on the
> tread are disabled.
> Will send log
>

**Comment by Jiffy176** [comment]
[debug-Sat_Mar_16_19_32_53_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14625026/debug-Sat_Mar_16_19_32_53_2024.log)
[debug-Sat_Mar_16_19_35_18_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14625031/debug-Sat_Mar_16_19_35_18_2024.log)
[debug-Sat_Mar_16_19_36_17_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14625035/debug-Sat_Mar_16_19_36_17_2024.log)

**Comment by Jiffy176** [comment]
Logs on way

**Comment by cagnulein** [comment]
got the issue. just a question: when you were pressing the plus inclination
tile, does the treadmill went to 1% and stops there right in these test?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle 20:42 Jiffy176 ***@***.***> ha
scritto:

**Comment by Jiffy176** [comment]
In the last test it did nothing.
Ones previously yes press incline plus and the tread stops.

**Comment by cagnulein** [comment]
ok as soon as i will have not my little daughter around i will check it.
probably monday

have a nice weekend

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 16 mar 2024 alle 21:02 Jiffy176 ***@***.***> ha
scritto:

> In the last test it did nothing.
> Ones previously yes press incline plus and the tread stops.
>

**Comment by Jiffy176** [comment]
You too Bro. Thank u

**Comment by cagnulein** [comment]
ok let's see if i did everything right this time https://github.com/cagnulein/qdomyos-zwift/actions/runs/8322652121
unfortunately i can't test it myself. keep the same setting
let me know!

**Comment by Jiffy176** [comment]
Sorry going mad I think I can't see an android build?

**Comment by cagnulein** [comment]
it's building, it will be ready at the bottom of the link in 10 minutes or so

**Comment by Jiffy176** [comment]
Ah cool thanks.
Whole new world to me 😊 thanks

**Comment by Jiffy176** [comment]
Ok so bluetooth now connects again but still the same issue that the BT
connection drops when the incline button is pressed in the app. Decline is
ok just the incline again.
Log to follow
S

**Comment by cagnulein** [comment]
could you please attach a debug log showing me this again? i would like to compare this scenario with the hci snoof. Thanks!

**Comment by Jiffy176** [comment]
[debug-Mon_Mar_18_09_33_02_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14633928/debug-Mon_Mar_18_09_33_02_2024.log)

So the scenario this time. 
Start tread manually and manual incline to 3
Then in app press decline and this time bt connection cut will now try what happens when I incline in app
Tx

**Comment by Jiffy176** [comment]
Same on incline BT connection disconnects

**Comment by cagnulein** [comment]
got the issue! i'm fixing it!

**Comment by Jiffy176** [comment]
Lol man you are a legend!!

**Comment by cagnulein** [comment]
hah wait you didn't enable the ""DKN Endurn Treadmill"" in this scenario. Please try again with this enable

**Comment by Jiffy176** [comment]
[debug-Mon_Mar_18_10_04_20_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14634397/debug-Mon_Mar_18_10_04_20_2024.log)

Now it doesn't connect via BT at all the symbol in the app goes blue but the machine doesn't register at all.

**Comment by cagnulein** [comment]
that's another issue, your treadmill is not sending its name over bluetooth
anymore. i suggest to restart it

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 18 mar 2024 alle ore 11:08 Jiffy176 ***@***.***>
ha scritto:

**Comment by Jiffy176** [comment]
[debug-Mon_Mar_18_10_13_24_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14634519/debug-Mon_Mar_18_10_13_24_2024.log)

So powered down. 
Counted to 10
Opened app. 
Plugged in treadmill
Same issue. 
App finds console and BT symbol goes blue but it is not connecting and machine not beeping when it connect. 
When I turn off dkn option connects ok
Just the incline issue then

**Comment by cagnulein** [comment]
ok got it and fixed https://github.com/cagnulein/qdomyos-zwift/actions/runs/8324935986
maybe we need another version to stabilize it, we're getting closed
remeber always to enable the dkn
let me know!

**Comment by Jiffy176** [comment]
Boom it works.
Do you want a log. Well done

**Comment by Jiffy176** [comment]
Does it need to be pushed to the app store, you said something about a 15 min something??
Thanks.

**Comment by cagnulein** [comment]
That's amazing! Yes I will push it to the play store soon. In the meantime, just write me an email and I will unlock it for you.

**Comment by Jiffy176** [comment]
What's your email pls

**Comment by cagnulein** [comment]
roberto.viola83@gmail.com

**Comment by cagnulein** [comment]
@Jiffy176 are we ok with this?

**Comment by Jiffy176** [comment]
Hi Roberto.
Yes I'm using the version you wrote for me a few weeks ago rather than the
app from the play store incase it hadn't been updated and it has been fine.
The only thing it does on autoincline is run ahead of the peloton class but
I'm not sure if there is anything that can be done about that or do I need
to set the setting to ignore the introduction  ?
Thanks.

**Comment by cagnulein** [comment]
that's great! for the time issue: are you pressing YES on QZ on the popup, after the intro of peloton?

**Comment by Jiffy176** [comment]
Yes

**Comment by Jiffy176** [comment]
Never new that was a thing

**Comment by cagnulein** [comment]
Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 17 apr 2024 alle ore 16:05 Jiffy176 ***@***.***>
ha scritto:

> Never new that was a thing
>
